### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by iky**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-matrix:
-  allow_failures:
-    - python: nightly
+env:
+  - NAMEKO_VERSION===2.8.5
+  - NAMEKO_VERSION===2.9.1
+  - NAMEKO_VERSION===2.10.0
+  - NAMEKO_VERSION===2.11.0
+  - NAMEKO_VERSION=>=2.12.0
+  - NAMEKO_VERSION===3.0.0-rc6
```
Comment (mattbennett) [inline .travis.yml]: Same comment as https://github.com/nameko/nameko-tracer/pull/33#discussion_r278009977

File: tests/api/test_dependency.py [hunk 2]
```diff
@@ -28,13 +29,27 @@ class Service(object):
...
+        try:
+            config_patch = nameko.config.patch(config)
+        except AttributeError:  # Nameko 2.X
+            config_patch = None
+            container = ServiceContainer(Service, config)
+        else:  # Nameko 3.X
+            config_patch.start()
+            container = ServiceContainer(Service)
+
+        containers.append((container, config_patch))
```
Comment (mattbennett) [inline tests/api/test_dependency.py]: Can you not use the `container_factory` fixture here?

File: .travis.yml (hunk 1 above)
Comment (iky) [reply .travis.yml]: :+1:

File: tests/api/test_dependency.py (hunk 2 above)
Comment (iky) [reply tests/api/test_dependency.py]: :+1: easier to use container_factory for the end-to-end tests. I also simplified the provider unit test so they don't need to construct the object this way anymore.

**Pull request merged**